conf/modules.config

```diff
@@ -4,201 +4,211 @@
+ *       publish_dir     = Directory to publish results.
+ *       publish_by_meta = Groovy list of keys available in meta map to append as directories to "publish_dir" path
+ *                         If publish_by_meta = true                 - Value of ${meta['id']} is appended as a directory to "publish_dir" path
+ *                         If publish_by_meta = ['id', 'custompath'] - If "id" is in meta map and "custompath" isn't then "${meta['id']}/custompath/"
+ *                                                                     is appended as a directory to "publish_dir" path 
+ *                         If publish_by_meta = false / null         - No directories are appended to "publish_dir" path
+ *       publish_files   = Groovy map where key = "file_ext" and value = "directory" to publish results for that file extension
+ *                         The value of "directory" is appended to the standard "publish_dir" path as defined above.
+ *                         If publish_files = null (unspecified)     - All files are published.
+ *                         If publish_files = false                  - No files are published.
+ *       suffix          = File name suffix for output files.
  *
  */
 
 params {
     modules {
         'sra_ids_to_runinfo' {
-            publish_dir   = 'public_data'
-            publish_files = ['tsv':'runinfo']
+            publish_dir     = 'public_data'
+            publish_files   = ['tsv':'runinfo']
         }
         'sra_runinfo_to_ftp' {
-            publish_dir   = 'public_data'
-            publish_files = ['tsv':'runinfo']
+            publish_dir     = 'public_data'
+            publish_files   = ['tsv':'runinfo']
         }
         'sra_fastq_ftp' {
-            publish_dir   = 'public_data'
-            publish_files = ['fastq.gz':'', 'md5':'md5']
+            publish_dir     = 'public_data'
+            publish_files   = ['fastq.gz':'', 'md5':'md5']
         }
         'sra_to_samplesheet' {
-            publish_dir   = 'public_data'
-            publish_files = false
+            publish_dir     = 'public_data'
+            publish_files   = false
         }
         'sra_merge_samplesheet' {
-            publish_dir   = 'public_data'
+            publish_dir     = 'public_data'
         }
         'gffread' {
-            args          = "--keep-exon-attrs -F -T"
-            publish_dir   = "genome"
+            args            = "--keep-exon-attrs -F -T"
+            publish_dir     = "genome"
         }
         'cat_fastq' {
-            publish_dir   = 'fastq'
+            publish_dir     = 'fastq'
         }
         'fastqc' {
-            args          = "--quiet"
+            args            = "--quiet"
+            // If meta = [id:WT_REP1, single_end:false, strandedness:reverse]
+            publish_by_meta = [ 'id', 'single_end', 'mytestfolder', 'strandedness']
+            // Output folder: results/fastqc/WT_REP1/single_end/mytestfolder/reverse/
```

Comment:
that would end up as 
`WT_REP1/true/`
and 
`WT_REP1/false`
wouldn't it? 

and do you really mean to merge the 'testfolder`?

Reply:
Yup, adding the `mytestfolder` in the path just allows developers to have additional separation of their folders if they want i.e. not just by the values in the `meta` map. Not sure how much it will be used but the code is flexible enough to deal with it.

Reply:
yes, but do we want that as default options of the rnaseq pipeline?

Reply:
Ah, no. I am going to close this PR. It was just a proof-of-principle. I will stick to using `publish_by_meta = true` in this pipeline to just publish by `meta.id` if required. The more complex use cases involving using a list of ids will probably be used in other, more complex pipelines like variant calling etc where you can have various levels of sample metadata.

Reply:
I knew I was missing something :smile: